flink-yarn-tests/src/test/java/org/apache/flink/yarn/testjob/YarnTestCacheJob.java

```diff
@@ -44,11 +44,16 @@
 public class YarnTestCacheJob {
 	private static final List<String> LIST = ImmutableList.of("test1", "test2");
 
-	public static JobGraph getDistributedCacheJobGraph() {
+	public static JobGraph getDistributedCacheJobGraph() throws IOException {
 		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
 
-		String cacheFilePath = Thread.currentThread().getContextClassLoader().getResource("cache.properties").getFile();
-
+		final String cacheFilePath = Thread.currentThread().getContextClassLoader()
+			.getResource("cache.properties").getFile();
+		final File testDirectory = File.createTempFile("test_directory", "");
+		testDirectory.delete();
+		testDirectory.mkdirs();
+		testDirectory.deleteOnExit();
```

Comment:
Let's factor this out into a helper method.

Reply:
Will pass a tmp directory of this method.